File: src/engine/commands/wieldcommand.cpp
#include "wieldcommand.h"

#include "util.h"


#define super Command

WieldCommand::WieldCommand(QObject *parent) :
    super(parent) {

    setDescription("Wield a weapon or shield that's in your inventory.\n"
                   "\n"
                   "Example: wield sword");
}

WieldCommand::~WieldCommand() {
}

void WieldCommand::execute(Player *player, const QString &command) {

    super::execute(player, command);

    if (!assertWordsLeft("Wield what?")) {
        return;
    }

    takeWord("(the|my)");

    GameObjectPtrList allItems = player->inventory();
    GameObjectPtr item = takeObject(allItems);
    if (!requireSome(item, "You don't have that.")) {
        return;
    }

    player->wield(item);
}

File: data/commands/wieldcommand.js
function WieldCommand() {
    Command.call(this);

    this.setDescription("Wield a weapon or shield that's in your inventory.\n" +
                        "\n" +
                        "Example: wield sword");
}

WieldCommand.prototype = new Command();
WieldCommand.prototype.constructor = WieldCommand;

WieldCommand.prototype.execute = function(player, command) {

    this.prepareExecute(player, command);

    this.takeWord("my");

    var item = this.takeObject(player.inventory);
    if (!this.requireSome(item, "Wield what?")) {
        return;
    }

    player.wield(item);
}

CommandRegistry.registerCommand("wield", new WieldCommand());
